src/mgos_onewire.c

```diff
@@ -276,18 +287,26 @@ void mgos_onewire_write_bytes(struct mgos_onewire *ow, const uint8_t *buf,
   }
 }
 
-struct mgos_onewire *mgos_onewire_create(int pin) {
+struct mgos_onewire *mgos_onewire_create_separate_io(int pin_in, int pin_out) {
   struct mgos_onewire *ow = calloc(1, sizeof(*ow));
   if (ow == NULL) return NULL;
-  ow->pin = pin;
-  if (!mgos_gpio_set_mode(ow->pin, MGOS_GPIO_MODE_INPUT) ||
-      !mgos_gpio_set_pull(ow->pin, MGOS_GPIO_PULL_UP)) {
+  ow->pin_in = pin_in;
+  ow->pin_out = pin_out;
+  if (!mgos_gpio_setup_input(ow->pin_in, MGOS_GPIO_PULL_UP)) {
     mgos_onewire_close(ow);
     return NULL;
   }
+  if (ow->pin_in != ow->pin_out) {
+    mgos_gpio_set_mode(ow->pin_out, MGOS_GPIO_MODE_OUTPUT);
```

Comment:
`mgos_gpio_setup_output(ow->pin_out, HIGH);`

Reply:
yes, but it is already done in the onewire_prepare_input. 
could replace that, but the real question is:
is the pin always considered to be captured after init? if not, then 
mgos_gpio_set_mode(ow->pin_out, MGOS_GPIO_MODE_OUTPUT);
should move to onewire_prepare_input. Otherwise i would consider it to be correct the way it is.

Reply:
> it is already done in the onewire_prepare_input
right, but it is less readable - after setting a mode ofr output pin we prepare an input pin for some reason?
i think `mgos_gpio_setup_output(ow->pin_out, HIGH);` makes the intention more clear: configure pin as output and set to high.

Reply:
> is the pin always considered to be captured after init?

not sure what you mean by this, can you clarify?

Reply:
sorry, i mean, during usage can the GPIO could be (theoretically) multiplexed (possibly also as input) in any scenario?; phrased otherwise: is it enough to call mgos_gpio_set_mode(ow->pin_out, MGOS_GPIO_MODE_OUTPUT); only once at creation?

Reply:
yes, we can assume we own the pins and nobody is messing with them behind our backs.